spec/models/miq_websocket_worker/runner_spec.rb

```diff
@@ -0,0 +1,26 @@
+describe MiqWebsocketWorker::Runner do
+  describe '#check_internal_thread' do
+    subject do
+      w = described_class.allocate
+      allow(w).to receive(:worker_initialization)
+      w.send(:initialize, :guid => MiqWebsocketWorker.create_worker_record.guid)
+      w
+    end
+
+    let(:worker) { double }
+    let(:app) { WebsocketServer.new }
+
+    it 'exits if the thread is not running' do
+      subject.instance_variable_set(:@worker, worker)
+      allow(worker).to receive(:rails_application).and_return(app)
+      expect(subject).to receive(:do_exit)
+
+      app.instance_variable_get(:@transmitter).kill
+      loop do
+        break if !app.instance_variable_get(:@transmitter).alive?
+        sleep 0.1
+      end
+      subject.check_internal_thread
```

Comment:
For a followup PR... I don't like the use of subject here as it I have to figure out what the subject is.  I think 

```ruby
let(:runner) do
  w = described_class.allocate
  allow(w).to receive(:worker_initialization)
  w.send(:initialize, :guid => MiqWebsocketWorker.create_worker_record.guid)
  w
end
```

Make this line more obvious:

```ruby
runner.check_internal_thread
```